crates/biome_js_analyze/src/lint/style/use_naming_convention.rs

```diff
@@ -450,15 +450,82 @@ declare_lint_rule! {
     /// - `formats`: the string [case] that the name must follow.
     ///   The supported cases are: [`PascalCase`], [`CONSTANT_CASE`], [`camelCase`], and [`snake_case`].
     ///
+    /// If only `formats` is set,
+    /// then `formats` is checked against the name of the declaration directly.
+    /// In the following configuration, we require `static readonly` class properties to be in [`CONSTANT_CASE`].
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///         "conventions": [
+    ///             {
+    ///                 "selector": {
+    ///                     "kind": "classProperty",
+    ///                     "modifiers": ["static", "readonly"]
+    ///                 },
+    ///                 "formats": ["CONSTANT_CASE"]
+    ///             }
+    ///         ]
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// The following code is then reported by the rule:
+    /// 
+    /// ```ts,use_options
+    /// class C {
+    ///     static readonly prop = 0;
+    /// }
+    /// ```
+    ///
+    /// A convention can shadow another one.
+    /// In the following configuration, the second convention is useless because it is always ignored.
```

Comment:
I suggest using plain language with consistency. It isn't clear what "shadow" means, let's use a more beginner friendly terminology.

In the second phrase, you use "ignored" instead of "shadow", but you don't explain why